### ISSUE TITLE
Create helper methods for file-level resources 

### ISSUE THREAD

**Issue by software-dov** [googleapis/gapic-generator-python#255]
Proto annotations allows for resources to be described at the file level instead of being a method option.

E.g. 
```proto
// Define a "User" resource owned by OS Login.
option (google.api.resource_definition) = {
  type: "oslogin.googleapis.com/User"
  pattern: "users/{user}"
};
```

These resources also need generated helper methods.

**Comment by software-dov** [comment]
See below for additional details
https://github.com/googleapis/googleapis/blob/d9576d95b44f64fb0e3da4760adfc4a24fa1faab/google/api/resource.proto#L35-L39

**Comment by software-dov** [comment]
Closed by #642
